### PR TITLE
Create STYLE_GUIDE.md

### PR DESCRIPTION
**Pull request by JafarAbdi**
### Description

Please explain the changes you made, including a reference to the related issue if applicable

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extended the tutorials / documentation, if necessary [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Include a screenshot if changing a GUI
- [ ] Document API changes relevant to the user in the moveit/MIGRATION.md notes
- [ ] Created tests, which fail without this PR [reference](http://docs.ros.org/kinetic/api/moveit_tutorials/html/doc/tests.html)
- [ ] Decide if this should be cherry-picked to other current ROS branches
- [ ] While waiting for someone to review your request, please consider reviewing [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

**Comment by JafarAbdi** [comment]
> @JafarAbdi Could you add that LOGGER should be inside the namespace by default? Also, we need to specify how loggers can be used in header files (there are some cases).

@henningkayser Adding loggers to header files could violate the one definition rule (for example two header files with the same namespace adding the logger to both headers will cause an error), do you mind sharing some cases so I could document them .?

**Comment by rhaschke** [comment]
> Adding loggers to header files could violate the one definition rule

For this reason, I suggest using a `static const` class variable named `LOGGER`, which then could be used/referenced from the header as well.

**Comment by JafarAbdi** [comment]
Some libraries just contain one source file, should the logger duplicate the name .?  for example `lma_kinematics_plugin` 

Should the logger name be `moveit_lma_kinematics_plugin.lma_kinematics_plugin` .? shouldn't `lma_kinematics_plugin` be sufficient .?

File: doc/STYLE_GUIDE.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# MoveIt2 style guide
```
Comment (davetcoleman) [inline doc/STYLE_GUIDE.md]: capitalize and add space before "2" per https://moveit.ros.org/about/press_kit/

File: doc/STYLE_GUIDE.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
```
Comment (davetcoleman) [inline doc/STYLE_GUIDE.md]: Add note that the [MoveIt 1 style guide](https://moveit.ros.org/documentation/contributing/code/) is in effect, except where noted here in this md file

File: doc/STYLE_GUIDE.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+## Logging
+
+Each source file should have a global static constant logger, we use <b>LOGGER</b> as the logger name with upper-case capitals, the logger should be inside the namespace by default. 
+
+The logger name consists of two parts <b>LIBRARY_NAME.SOURCE_NAME</b> e.g.
+
+`static const rclcpp::Logger LOGGER = rclcpp::get_logger("LIBRARY_NAME.SOURCE_NAME");`
+
+You could find the `LIBRARY_NAME` inside the `CMakeLists.txt` file of a package `set(MOVEIT_LIB_NAME LIBRARY_NAME)`
```
Comment (davetcoleman) [inline doc/STYLE_GUIDE.md]: Is there anything beyond Logging we should note here? Like how we do CMakeLists?

@henningkayser seems the decision from [here](https://github.com/ros-planning/moveit2/issues/22) belongs in this doc

**Pull request merged**